build-tools-internal/src/main/java/org/elasticsearch/gradle/internal/ElasticsearchJavaBasePlugin.java

```diff
@@ -112,6 +123,10 @@ public static void configureCompile(Project project) {
         java.setSourceCompatibility(BuildParams.getMinimumRuntimeVersion());
         java.setTargetCompatibility(BuildParams.getMinimumRuntimeVersion());
         project.getTasks().withType(JavaCompile.class).configureEach(compileTask -> {
+            compileTask.getJavaCompiler().set(javaToolchains.compilerFor(spec -> {
+                spec.getLanguageVersion().set(JavaLanguageVersion.of(BuildParams.getMinimumRuntimeVersion().getMajorVersion()));
```

Comment:
Should we actually use `minimumCompilerVersion`? Although, I struggle to see why we have both, and when they would ever be different.

Reply:
I was mimicking the source and target compatibility above. It does seem odd that we have both.

Reply:
Yeah, on the 7.17 branch, the `minimumCompilerVersion` is 17 while `minimumRuntimeVersion` is 8. I believe the reason for this is because to keep the build logic consistent across these branches we require Java 17 to build 7.17 even those the production code targets Java 8. In reality it should be `minimumGradleJavaVersion` but we can follow up to clean this stuff up.